iOS-TvShows/iOS-TvShows/Modules/Shows/Components/TableViewCell/ShowTableViewCell.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  ShowTableViewCell.swift
+//  iOS-TvShows
+//
+//  Created by Filip Ban on 24.11.2021..
+//
+
+import UIKit
+import Kingfisher
+import RxCocoa
+
+struct ShowTableCellItem: TableCellItem {
+    let show: Show
+    let didSelect: ((ShowTableCellItem) -> Void)?
+}
+
+extension ShowTableCellItem {
+    func cell(from tableView: UITableView, at indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(ofType: ShowTableViewCell.self, for: indexPath)
+        cell.setData(with: show)
+        return cell
+    }
+
+    var height: CGFloat {
+        114
+    }
+
+    func didSelect(at indexPath: IndexPath) {
+        didSelect?(self)
+    }
+}
+
+class ShowTableViewCell: UITableViewCell {
+
+    var customImageView: UIImageView!
+    var label: UILabel!
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        backgroundColor = UIColor.TVShows.appWhite
+        customImageView = UIImageView()
+        contentView.addSubview(customImageView)
+        label = UILabel(with: UIFont.systemFont(ofSize: 17))
+        contentView.addSubview(label)
+
+        customImageView.clipsToBounds = true
+        customImageView.roundAllCorners(withRadius: 4)
+
+        customImageView.snp.makeConstraints {
+            $0.size.equalTo(CGSize(width: 64, height: 90))
+            $0.centerY.equalToSuperview()
+            $0.leading.equalToSuperview().inset(16)
+        }
+
+        label.snp.makeConstraints {
+            $0.leading.equalTo(customImageView.snp.trailing).offset(24)
+            $0.trailing.equalToSuperview().inset(24)
+            $0.centerY.equalToSuperview()
+        }
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
```

Comment:
Not sure why fatalError here? You are setting up the view from code, why just not reuse the same code here as well? Something like this:

```swift
    override init(frame: CGRect) {
        super.init(frame: frame)
        setupLayout()
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        setupLayout()
    }

    /// Method for layouting views from code. The method does nothing as is,
    /// but it is called during the layout cycle has should be used for layouting
    /// your view from code. In short, override this method with your layout code.
    /// You can call super but it is optional.
    func setupLayout() {
        // Method is used for overriding
    }
```

Reply:
just a habit